cabal-install/Distribution/Client/RebuildMonad.hs

```diff
@@ -166,3 +176,58 @@ monitorDirectoryStatus dir = do
                     then monitorDirectory dir
                     else monitorNonExistentDirectory dir]
 
+doesFileExistMonitored :: FilePath -> Rebuild Bool
+doesFileExistMonitored f = do
+    root <- askRoot
+    exists <- liftIO $ doesFileExist (root </> f)
+    -- TODO: If the file exists, should we really monitor the entire
+    -- file?!
```

Comment:
Ahh! Well, funny you should ask for it. Internally this is supported (and is used for globs), it just wasn't exposed for individual files since we didn't have a use case (and I couldn't think of one).

So just add to the FileMonitor module:
```
-- | Monitor a single file for existence only. The monitored file is
-- considered to have changed if it no longer exists.
--
monitorFileExistence :: FilePath -> MonitorFilePath
monitorFileExistence = MonitorFile FileExists DirNotExists
```